pkg/monitortests/network/disruptionpodnetwork/monitortest.go

```diff
@@ -202,11 +202,48 @@ func (pna *podNetworkAvalibility) StartCollection(ctx context.Context, adminREST
 		if _, err = pna.kubeClient.AppsV1().Deployments(pna.namespaceName).Create(context.Background(), deployment, metav1.CreateOptions{}); err != nil {
 			return err
 		}
+
+		// The pods need to go active and the pollers wait until we have at least one before starting.
+		checkForPods := func(ctx context.Context) (bool, error) {
+			return pna.podsAreReady(ctx, deployment)
+		}
+		err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 120*time.Second, true, checkForPods)
+		if err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
+// Checks if the Pods are ready for the given deployment
+func (pna *podNetworkAvalibility) podsAreReady(ctx context.Context, deployment *appsv1.Deployment) (bool, error) {
```

Comment:
to check if a deployment is ready, use the .status.AvailableReplicas and compare against .spec.Replicas.